native/machine-coding/quiz-app-with-timer/index.js

```diff
@@ -9,52 +9,52 @@ const restart = document.getElementById("restart");
 const userScore = document.getElementById("user-score");
 const startScreen = document.querySelector(".start-screen");
 const startButton = document.getElementById("start-button");
-let questionCount = 0; // Initialize questionCount
+let questionCount = 0;
 let scoreCount = 0;
 let count = 11;
 let countdown;
+let selectedQuestions; // Declare selectedQuestions variable
 
 // Quiz questions array
 let quizArray;
 
-// Fetch the questions from the JSON file
+// Function to shuffle and select random questions
+function selectRandomQuestions(questions, count) {
+  const shuffledQuestions = questions.slice().sort(() => Math.random() - 0.5);
+  return shuffledQuestions.slice(0, count);
+}
+
+// Fetch the questions from the JSON file and initialize the quiz
 fetch("questions.json")
-    .then((response) => response.json())
-    .then((data) => {
-        quizArray = data;
-        quizCreator();
-        initial(); // Initialize the quiz
-    })
-    .catch((error) => {
-        console.error("Error loading questions:", error);
-    });
+  .then((response) => response.json())
+  .then((data) => {
+    quizArray = data;
+    initial(); // Initialize the quiz with random questions
+  })
+  .catch((error) => {
+    console.error("Error loading questions:", error);
+  });
 
-//Restart Quiz
+// Restart Quiz
 restart.addEventListener("click", () => {
   initial();
   displayContainer.classList.remove("hide");
   scoreContainer.classList.add("hide");
 });
 
-//Next Button
+// Next Button
 nextBtn.addEventListener(
   "click",
   (displayNext = () => {
-    //increment questionCount
     questionCount += 1;
-    //if last question
-    if (questionCount == quizArray.length) {
-      //hide question container and display score
+    if (questionCount === selectedQuestions.length) {
       displayContainer.classList.add("hide");
       scoreContainer.classList.remove("hide");
-      //user score
       userScore.innerHTML =
-        "Your score is " + scoreCount + " out of " + questionCount;
+        "Your score is " + scoreCount + " out of " + selectedQuestions.length;
     } else {
-      //display questionCount
       countOfQuestion.innerHTML =
-        questionCount + 1 + " of " + quizArray.length + " Question";
-      //display quiz
+        questionCount + 1 + " of " + selectedQuestions.length + " Question";
       quizDisplay(questionCount);
       count = 11;
       clearInterval(countdown);
```

Comment:
The timer bug is not fixed yet. You probably need to reset the HTML of the timer along with these other things.

Reply:
I dont know why but i am really unable to recreate the issue you are facing.
Can you provide the steps to recreate the bug.

Reply:
The old timer is shown for 1 sec after going to the next question

Reply:
ok i will fix it but what about that 8s and 10s issue?

Reply:
Yes thats the same issue